### PR TITLE
feat: Apache Spark on Amazon Athena - `wr.athena.create_spark_session` & `wr.athena.run_spark_calculation`

### PR DESCRIPTION
**Pull request by kukushking**
### Feature or Bugfix
- Feature

### Scope
- This PR adds basic capabilities to start a session and kick off a calculation. Any potential data integration (i.e. ability to pass data frames, read results) will be addressed in next PRs.

### Detail
- Add `create_spark_session` and `run_spark_calculation` with corresponding waiters
- Add simple test case
- Expand test case
- Add tutorial / docstrings
- Add test infra IAM role

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: awswrangler/athena/_spark.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    session_id: Optional[str] = None,
+    notebook_id: Optional[str] = None,
+    notebook_version: str = "Athena notebook version 1",
+    coordinator_dpu_size: int = 1,
+    max_concurrent_dpus: int = 5,
+    default_executor_dpu_size: int = 1,
+    additional_configs: Optional[Dict[str, Any]] = None,
+    idle_timeout: int = 15,
+    boto3_session: Optional[boto3.Session] = None,
+) -> "GetCalculationExecutionResponseTypeDef":
```
Comment (LeonLuttenberger) [inline awswrangler/athena/_spark.py:161]: So far, we haven't been returning any of the boto3 stub types from our public facing functions. Our reasoning was that for customers who don't use the boto3 stubs, it'll be more helpful for the return type to be returned as something like `Dict[str, Any]` than just `Any`.

File: awswrangler/athena/_spark.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        else session_id
+    )
+
+    response = client_athena.start_calculation_execution(
+        SessionId=session_id,
+        CodeBlock=code,
+    )
+    _logger.info("Calculation execution info:\n%s", response)
+
+    return _get_calculation_execution_results(
```
Comment (LeonLuttenberger) [inline awswrangler/athena/_spark.py:207]: Can the return value of an Athena Spark execution be a DataFrame? Or will the output always just be written to an S3 location?

I'm mainly just wondering if there's a way to sensibly make these API calls accept or return Data Frames.

File: awswrangler/athena/_spark.py (hunk 1 above)
Comment (kukushking) [reply awswrangler/athena/_spark.py:161]: Yeah good point, will change back to dicts

File: awswrangler/athena/_spark.py (hunk 2 above)
Comment (kukushking) [reply awswrangler/athena/_spark.py:207]: I am looking into the ways to support that where it is applicable but that is very much dependent on what spark code you are running. 

There is json metadata file in the results path along with stdout/err text files, but it's almost always empty, well at least have been in my tests so far. Documentation is a bit lacking on this. I'll play around to see what we can make of it.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: b6f3244a418a368c86cbcca0e751a8bb44a4b29a
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=58df02e6-9be9-4e16-8e93-04d9a853cf6b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: ad102148fa08f3417b4fef16ebf0f47969036b22
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=cf7d5a44-f582-43ca-bcdf-efe6f6281855%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: ad102148fa08f3417b4fef16ebf0f47969036b22
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e9484e37-cda0-4836-b322-8043ecc23b80%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: c48cfd973ccdbbd3b7808d7ad76f7aa437df5b8b
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8e230387-3202-4a49-80b8-e3f77782806d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: c48cfd973ccdbbd3b7808d7ad76f7aa437df5b8b
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7baaa21e-5dc5-40b5-a025-3ae647d4e1c6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: test_infra/stacks/base_stack.py [hunk 3]
```diff
@@ -127,6 +127,27 @@ def __init__(self, scope: Construct, construct_id: str, **kwargs: str) -> None:
...
+            ],
+            inline_policies={
+                "GetDataAccess": iam.PolicyDocument(
+                    statements=[
+                        iam.PolicyStatement(
+                            actions=["lakeformation:GetDataAccess"],
+                            resources=["*"],
+                        ),
+                    ]
+                ),
```
Comment (jaidisido) [inline test_infra/stacks/base_stack.py:148]: Nitpick - I feel like these managed policies and LF GetDataAccess inline policy is specified 3-4 times in this base stack, might be worth putting it in a dict and passing it around to reduce LOC?

**Review by jaidisido — COMMENTED** [review]
Looks great, a couple of minor comments

File: test_infra/stacks/base_stack.py (hunk 3 above)
Comment (kukushking) [reply test_infra/stacks/base_stack.py:148]: Agreed

**Review by jaidisido — APPROVED** [review]

**Review by LeonLuttenberger — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: e062ab8fd89e3f26afa76ca9e13eed8d8f672545
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=67bb859f-6afd-4d84-90c5-8a8273e68c35%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**